jni/CMakeLists.txt

```diff
@@ -143,12 +147,20 @@ if (${CONFIG_FAISS} STREQUAL ON OR ${CONFIG_ALL} STREQUAL ON OR ${CONFIG_TEST} S
     endif ()
 
     # Check if patch exist, this is to skip git apply during CI build. See CI.yml with ubuntu.
-    find_path(PATCH_FILE NAMES 0001-Custom-patch-to-support-multi-vector.patch PATHS ${CMAKE_CURRENT_SOURCE_DIR}/patches/faiss NO_DEFAULT_PATH)
+    find_path(PATCH_FILE NAMES 0001-Custom-patch-to-support-multi-vector.patch 0002-Custom-patch-to-support-sqfp16-neon.patch PATHS ${CMAKE_CURRENT_SOURCE_DIR}/patches/faiss NO_DEFAULT_PATH)
 
     # If it exists, apply patches
     if (EXISTS ${PATCH_FILE})
         message(STATUS "Applying custom patches.")
         execute_process(COMMAND git apply --ignore-space-change --ignore-whitespace --3way ${CMAKE_CURRENT_SOURCE_DIR}/patches/faiss/0001-Custom-patch-to-support-multi-vector.patch WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/external/faiss ERROR_VARIABLE ERROR_MSG RESULT_VARIABLE RESULT_CODE)
+
+        # 0002-Custom-patch-to-support-sqfp16-neon.patch is a temporary patch to add NEON support to SQ.
+        # Once the commit conflict issues wrt to Multi vector are resolved, this patch can be removed by updating the faiss submodule with corresponding commit.
+        # Apply the patch if the OS is not Windows and Processor is aarch64.
+        if (NOT WIN32 AND ${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64" AND ${ENABLE_SIMD})
```

Comment:
Why do we want this patch to be platform dependent? This will be applied always if we update faiss submodule anyway in the future.

Reply:
It has NEON changes. So, we wan to apply it only on aarch64 and when SIMD is enabled

Reply:
There is no harm to apply the patch for other platform I believe? Later, if we just update the faiss commit number, how can we control to include that commit only for aarch64 and when SIMD is enabled?

Reply:
There might be no harm to apply it on x86. But, we still need to apply only SIMD is enabled to avoid crashing the cluster on some processors which don't support NEON. 

After updating the commit version, we might need to create a patch to revert the changes if SIMD is not enabled

Reply:
Based on the comment, that was not a plan before. We were going to simply remove this custom patch later.
Also, reverting a commit from faiss based on parameter doesn't seems ideal way to handle the issue?

Reply:
Yes Heemin, I understand that is not an ideal approach. But, it will be a breaking change for the systems that don't support SIMD. We will remove this compatibility from the next major release, until then we need to do this.

Reply:
@heemin32 I think there was some misunderstanding and things are not clear for you.
1. Even for x86 we are not enabling SIMD by default. That's the reason we are adding this optional parameter `enable_SIMD`. To enable AVX2 we need to set this to `true`, which is by default `false`. 
2. AFAIK, NEON is enabled by default on `aarch64`([Reference](https://stackoverflow.com/questions/29851128/gcc-arm64-aarch64-unrecognized-command-line-option-mfpu-neon/29891469#29891469)). So, there is no way to control it explicitly like we do for AVX2

Reply:
When the neon code runs in machine which does not support SIMD, does it crash even if user does not use the sqfp16? If it does not crash as long as the neon code is not executed, another option is that we can block the sqfp16 using setting for arm and let the user to enable it if needed.

Also, do you know any aarch64 platform which does not support SIMD? I could not find any. Even if there is one, I wonder if there are any user running OpenSearch on it. It seems aarch64 should support SIMD as standard? https://en.wikipedia.org/wiki/AArch64

Reply:
Talked offline. We are going we patch approach to control SIMD in arm.